examples/firewall/firewall.c

```diff
@@ -0,0 +1,388 @@
+                                RTE_LOG(INFO, APP, "Packet from source IP %d has been dropped.\n", ipv4_hdr->src_addr);
+                        }
+                } else {
+                        switch (rule) {
+                                case 0:
+                                        meta->action = ONVM_NF_ACTION_TONF;
+                                        meta->destination = destination;
+                                        stats.pkt_accept++;
+                                        stats.pkt_total++;
+                                        if (debug) {
+                                                RTE_LOG(INFO, APP, "Packet from source IP %d has been accepted.\n",
+                                                        ipv4_hdr->src_addr);
+                                        }
+                                        break;
+                                default:
+                                        meta->action = ONVM_NF_ACTION_DROP;
+                                        stats.pkt_drop++;
+                                        stats.pkt_total++;
+                                        if (debug) {
+                                                RTE_LOG(INFO, APP, "Packet from source IP %d has been dropped.\n",
+                                                        ipv4_hdr->src_addr);
+                                        }
+                                        break;
+                        }
+                }
+        } else {
+                if (debug) {
+                        RTE_LOG(INFO, APP, "Packet received not ipv4\n");
+                }
+                stats.pkt_drop++;
+                stats.pkt_total++;
+                meta->action = ONVM_NF_ACTION_DROP;
+        }
+
+        if (++counter == print_delay) {
+                do_stats_display();
+                counter = 0;
+        }
+
+        return 0;
+}
+
+static int lpm_setup(struct onvm_fw_rule **rules, int num_rules) {
+        int i, status;
+
+        firewall_req = (struct lpm_request *) rte_malloc(NULL, sizeof(struct lpm_request), 0);
+
+        if (!firewall_req) return 0;
+
+        firewall_req->max_num_rules = 1024;
+        firewall_req->num_tbl8s = 24;
+        firewall_req->socket_id = rte_socket_id();
+        firewall_req->name[0] = 'f';
+        firewall_req->name[1] = 'w';
+
+        status = onvm_nflib_request_lpm(firewall_req);
+
+        if (status < 0) {
+                rte_exit(EXIT_FAILURE, "Cannot get lpm region for firewall\n");
+        }
+
+        lpm_tbl = rte_lpm_find_existing("fw");
+
+        if (lpm_tbl == NULL) {
+                printf("No existing LPM_TBL\n");
+        }
+
+        for (i = 0; i < num_rules; ++i) {
+                printf("RULE %d: { ip: %d, depth: %d, action: %d }\n", i, rules[i]->src_ip, rules[i]->depth,
+                       rules[i]->action);
+                int add_failed = rte_lpm_add(lpm_tbl, rules[i]->src_ip, rules[i]->depth, rules[i]->action);
```

Comment:
Prefer a generic `ret = rte_lpm_add...`, this was we don't have extra definitions and its just cleaner